cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js

```diff
@@ -446,6 +446,86 @@ class HydratorPlusPlusNodeConfigCtrl {
     }
     return true;
   }
+  validatePluginPreferences() {
+    const nodeInfo = this.state.node;
+    let schemaParseError = null;
+    let vm = this;
+    vm.validating = true;
+    const pluginInfo = angular.copy(nodeInfo.plugin);
+    pluginInfo.type = nodeInfo.type;
+
+    const requestBody = {
+      stage: {
+        name: nodeInfo.name,
+        plugin: pluginInfo,
+      },
+      inputSchemas: !nodeInfo.inputSchema ? [] : nodeInfo.inputSchema.map((input) => {
+        let schema;
+        try {
+          schema = JSON.parse(input.schema);
+        } catch (e) {
+          schemaParseError = e;
+        }
+        return {
+          stage: input.name,
+          schema
+        };
+      })
+    };
+
+    if (schemaParseError) {
+      vm.validationErrors = [schemaParseError];
+      return;
+    }
+
+    const parseResSchema = (res) => {
+      if (res.name && res.type && res.fields) {
+        return [this.HydratorPlusPlusNodeService.getOutputSchemaObj(res)];
+      }
+      let schemaArr = [];
+      angular.forEach(res, (value, key) => {
+        if (value.name && value.type && value.fields) {
+          schemaArr.push(this.HydratorPlusPlusNodeService.getOutputSchemaObj(value, key));
+        }
+      });
+      let recordSchemas = schemaArr.filter(schema => schema.name.substring(0, 6) === 'record');
+      let schemaArrWithoutRecordSchemas = _.difference(schemaArr, recordSchemas);
+      let schemaArrWithSortedRecordSchemas = schemaArrWithoutRecordSchemas.concat(_.sortBy(recordSchemas, 'name'));
+      return schemaArrWithSortedRecordSchemas;
+    };
+
+    const params = {
+      context: this.$state.params.namespace
+    };
+
+    this.myPipelineApi.validateStage(params, requestBody)
+      .$promise
+      .then((res) => {
+        vm.validating = false;
+        if (res.errors.length > 0) {
+          let errors = res.errors.map(err => err.message);
+          vm.validationErrors = errors;
+        } else {
+          const outputSchema = this.myHelpers.objectQuery(res, 'spec', 'outputSchema');
+          const portSchemas = this.myHelpers.objectQuery(res, 'spec', 'portSchemas');
+          if (!outputSchema && !portSchemas) {
+            vm.validationErrors = ['There is no output schema.'];
```

Comment:
I think no output schema is actually a valid case

Reply:
Stage validation in output-schema.js considers this an error, I can make changes to make this a valid case, can you confirm?
https://github.com/cdapio/cdap/blob/b31d0c456088719e7bfbaad67b9973b8441ae523/cdap-ui/app/directives/plugin-functions/functions/output-schema/output-schema.js#L142

Reply:
I think it is actually a valid case. Just empty out the output schema.